core/src/test/java/com/datastax/oss/driver/internal/core/protocol/CompressSubstitutionsTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.protocol;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.datastax.oss.driver.shaded.guava.common.collect.Sets;
+import com.datastax.oss.driver.shaded.guava.common.collect.Sets.SetView;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class CompressSubstitutionsTest {
```

Comment:
OK to keep the test then, but the current test is way too complex, and method names do not comply with our guidelines.

Suggested change:

```
@RunWith(DataProviderRunner.class)
public class CompressSubstitutionsTest {

  private static final Set<String> EXCLUDED_METHOD_NAMES =
      ImmutableSet.of("algorithm", "readUncompressedLength");

  private static final Comparator<Method> METHOD_SIGNATURE_COMPARATOR =
      Comparator.comparing(Method::getName)
          .thenComparing(Method::getReturnType, (ret1, ret2) -> ret1.equals(ret2) ? 0 : 1)
          .thenComparing(
              Method::getParameterTypes,
              (params1, params2) -> Arrays.deepEquals(params1, params2) ? 0 : -1);

  @Test
  @UseDataProvider(value = "substitutionClasses")
  public void should_substitute_compressor_methods(
      Class<Lz4Compressor> substitutedClass, Class<Lz4Substitution> substitutionClass) {

    List<Method> methodsToSubstitute =
        Arrays.stream(substitutedClass.getDeclaredMethods())
            .filter(method -> !EXCLUDED_METHOD_NAMES.contains(method.getName()))
            .collect(Collectors.toList());

    List<Method> substitutedMethods =
        Arrays.stream(substitutionClass.getDeclaredMethods()).collect(Collectors.toList());

    assertThat(methodsToSubstitute)
        .usingElementComparator(METHOD_SIGNATURE_COMPARATOR)
        .isEqualTo(substitutedMethods);
  }

  @DataProvider
  public static Object[][] substitutionClasses() {
    return new Object[][] {
      {Lz4Compressor.class, Lz4Substitution.class},
      {SnappyCompressor.class, SnappySubstitution.class}
    };
  }
}```

Reply:
done, thx